vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
I think I'd like to see the `authentication_component` method in `AuthenticationMixin` take an array of parameters to check.  That way, the caller (`has_credentials`, I believe) would be able to check `authentication_component(type, [:userid, :password])`.

@Fryguy, @jrafanie: thoughts?  Not sure if this would massively break the rest of the world...